### PR TITLE
Handle closing stream sessions correctly when the header has the fin flag set

### PR DESCRIPTION
**Pull request by madster26**
Currently when the header frame comes with the FIN_FLAG, whiskey does not complete the request because it checks for isClosed() which verifies if the stream is closed both remotely & locally. 

This fix closes the stream remotely when it finds the FIN_FLAG set because there should be no more frames that should be received on this stream. 

Also the readHeadersEnd method checks if the stream is closed Remotely and completes the request.

### CODE REVIEW

File: src/main/java/com/twitter/whiskey/net/SpdySession.java [hunk 1]
```diff
@@ -497,7 +497,9 @@ public void readHeadersFrame(int streamId, boolean last) {
 
         if (stream == null || stream.isClosedRemotely()) {
             sendRstStream(streamId, SPDY_STREAM_INVALID_STREAM);
-        }
+        } else if (last) {
+ 					  stream.closeRemotely();
```
Comment (yschimke) [inline src/main/java/com/twitter/whiskey/net/SpdySession.java]: janky formatting

Comment (madster26) [inline src/main/java/com/twitter/whiskey/net/SpdySession.java]: Ugh vi!!!!

**Comment by kgoodier** [comment]
What's up with this pull request? Shows a whole bunch of commits, but under "Files Changed" it only has  the "Handle closing stream sessions" one (6365911).

**Comment by madster26** [comment]
@kgoodier I think it is just showing all my previous commits to this repo for some reason. If you think this might not work let me know I can drop this PR and create a new one.

**Comment by kgoodier** [comment]
When I look at "Files Changed", I see 2 lines changed, and they look good. I'm concerned that merging this PR may grab a whole bunch of commit history that doesn't really apply, though, especially since I don't know why Github is showing it. Would you mind making this change again in a new commit on a clean branch and submitting a PR? Thanks!

**Comment by madster26** [comment]
Created PR 20

**Pull request closed without merging**